shared/src/commonMain/kotlin/com/devil/phoenixproject/util/DataBackupManager.kt

```
@@ -449,11 +310,34 @@ abstract class BaseDataBackupManager(
+            // Count of per-entity failures that were logged and skipped instead of
+            // aborting the whole import. Surfaced to the UI so users can see partial
+            // data loss rather than a silent pass.
+            var entitiesWithErrors = 0
+
+            // Helper: run a single-entity write inside the outer transaction. If it
+            // throws, log the cause + entity identifier and continue with the next
+            // entity. SQLDelight's transaction block tolerates caught exceptions —
+            // only an uncaught throwable rolls the outer transaction back.
+            fun <T> tryImport(label: String, entityId: String?, block: () -> T): T? = try {
+                block()
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)